doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+    {
+        InstalledPackages,
+        /// IMPLEMENTATION NOTE:  Windows Package Manager does not currently have this concept. 
+        /// This would be a new Orchestration feature to keep track of the installing CatalogPackage objects and allow 
+        /// search of those objects.
+        InstallingPackages,
+        AllLocalPackages
+    };
+
+    /// 
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass GetCompositeAppCatalogOptions
+    {
+        GetCompositeAppCatalogOptions(); 
+
+        /// Get a composite catalog to allow searching a user defined or pre defined source
+        /// and a local source (Installing, Installed) together  
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports 
+        /// one local source and one remote source per composite catalog.
+        /// DESIGN NOTE: It seems likely that in the future we will want to support 
+        /// letting callers use two local sources (i.e. both Installing and Installed)
+        IVector<AppCatalog> Catalogs { get; };
+        /// The local source(s)
+        LocalAppCatalog LocalAppCatalog;
+        /// Sets the default search behavior if the catalog is a composite catalog.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallScope
+    {
+        User,
+        Machine,
+    };
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallMode
+    {
+        /// The default experience for the installer. Installer may show some UI.
+        Default,
+        /// Runs the installer in silent mode. This suppresses the installer's UI to the extent 
+        /// possible (installer may still show some required UI).
+        Silent,
+        /// Runs the installer in interactive mode.
+        Interactive,
+    };
+
+    /// Options when installing a package.
+    /// Intended to allow full compatibility with the "winget install" command line interface.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallOptions
+    {
+        InstallOptions();
+
+        /// DESIGN NOTE: --Exact from the winget install command line is implied here by virtue of the find 
+        /// then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up 
+        /// front in order to get the CatalogPackage object.
+        /// DESIGN NOTE: Using PackageVersionInfo rather than CatalogPackage felt like it would be too 
+        /// specific here. CatalogPackage and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about 
+        /// which version is preferred.
+        CatalogPackage CatalogPackage;

```

Comment:
Is this a mandatory parameter? (I.e., not really an "option".)

Reply:
Yes.